public/app/plugins/datasource/loki/queryHints.ts

```diff
@@ -0,0 +1,42 @@
+import { DataFrame, QueryHint } from '@grafana/data';
+
+import { isQueryWithParser } from './query_utils';
+import { extractLogParserFromDataFrame } from './responseUtils';
+
+export function getQueryHints(query: string, series: DataFrame[]): QueryHint[] {
+  const hints: QueryHint[] = [];
+  if (series.length > 0) {
+    const { hasLogfmt, hasJSON } = extractLogParserFromDataFrame(series[0]);
+    const queryWithParser = isQueryWithParser(query);
+
+    if (hasJSON && !queryWithParser) {
+      hints.push({
+        type: 'ADD_JSON_PARSER',
+        label: 'Selected log stream selector has JSON formatted logs. ',
```

Comment:
Just a nit, but I think the last whitespace can be removed, as there is already a whitespace between this label and the `fix.label` in QueryBuilderHints.tsx.

Reply:
👍